main.py

```diff
@@ -313,6 +315,100 @@ async def get_running_experiment() -> Optional[int]:
     return None
 
 
+def organize_result_directory(result_dir_path: str, rid: str) -> bool:
+    """Organizes the result directory.
+    
+    It performs the following:
+        1. Read the metadata file from the RID directory and remove it.
+        2. Move the modified experiment to the RID directory.
+        3. Find the h5 result file and copy it to the RID directory.
+        4. Dump the metadata on the copied result file.
+
+    Args:
+        result_dir_path: The full path of the result directory.
+        rid: The RID string.
+
+    Returns:
+        True if the h5 result file exists.
+        Otherwise, stop organizing the result directory and return False. 
+    """
+    rid_dir_path = posixpath.join(result_dir_path, f"{rid}/")
+    # read and remove the metadata
+    metadata_path = posixpath.join(rid_dir_path, "metadata.json")
+    with open(metadata_path, encoding="utf-8") as metadata_file:
+        metadata = json.load(metadata_file)
+    submission_time_str, visualize = metadata["submission_time"], metadata["visualize"]
+    submission_time = datetime.fromisoformat(submission_time_str)
+    date = submission_time.date().isoformat()
+    hour = submission_time.hour
+    # find the result file
+    datetime_result_dir_path = posixpath.join(result_dir_path, f"{date}/{hour}/")
+    padded_rid = rid.zfill(9)
+    for item in os.listdir(datetime_result_dir_path):
+        if item.startswith(padded_rid):
+            break
+    else:  # no result file
+        return False
+    # copy the result file to the RID directory
+    result_path = posixpath.join(datetime_result_dir_path, item)
+    copied_result_path = posixpath.join(rid_dir_path, "result.h5")
```

Comment:
The term "copied" is quite ambiguous: if you copy A and get B, we can say both A and B are _copied_.
Actually I think it is more common to say that "A is copied" and "B is a copy of A".

Therefore, I would suggest `new_result_path` for example. Or, you can use "src" and "dst" which are popular terms in computer science (`shutil` documentation also uses those terms).